Sources/Core/Particle/ParticleSystemData.cpp

```
@@ -422,12 +422,14 @@ void ParticleSystemData<N>::Deserialize(const std::vector<uint8_t>& buffer)
         GetFlatbuffersParticleSystemData<N>::GetParticleSystemData(
             buffer.data());
     Deserialize(fbsParticleSystemData, *this);
+    Resize(NumberOfParticles());
 }
 
 template <size_t N>
 void ParticleSystemData<N>::Set(const ParticleSystemData& other)
 {
     *this = other;
```

Cppcheck (2.21.0)

[warning] The address of variable 'x' might be accessed at non-zero index.

(objectIndex)